repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/inspectors/metrics4bears/TestMetrics4BearsJsonFile.java

```diff
@@ -0,0 +1,435 @@
+        GlobalStepInfo building = reproductionBuggyBuild.getProcessDurations().getBuilding();
+        List<String> expectedBuildingStepNames = new ArrayList<>(Arrays.asList("CheckoutPatchedBuildCandidate",
+                "ComputeSourceDir", "ComputeTestDir", "CheckoutBuggyBuildCandidateSourceCode",
+                "BuildProjectBuggyBuildCandidateSourceCode"));
+        List<String> buildingStepNames = building.getStepNames();
+        assertThat(building.getNbSteps(), is(expectedBuildingStepNames.size()));
+        assertThat(building.getStepNames().size(), is(expectedBuildingStepNames.size()));
+        assertThat(building.getStepDurations().size(), is(expectedBuildingStepNames.size()));
+        assertThat(buildingStepNames.containsAll(expectedBuildingStepNames), is(true));
+
+        GlobalStepInfo testing = reproductionBuggyBuild.getProcessDurations().getTesting();
+        List<String> expectedTestingStepNames = new ArrayList<>(Arrays.asList("TestProjectBuggyBuildCandidateSourceCode"));
+        List<String> testingStepNames = testing.getStepNames();
+        assertThat(testing.getNbSteps(), is(expectedTestingStepNames.size()));
+        assertThat(testing.getStepNames().size(), is(expectedTestingStepNames.size()));
+        assertThat(testing.getStepDurations().size(), is(expectedTestingStepNames.size()));
+        assertThat(testingStepNames.containsAll(expectedTestingStepNames), is(true));
+
+        GlobalStepInfo fixing = reproductionBuggyBuild.getProcessDurations().getFixing();
+        List<String> expectedFixingStepNames = new ArrayList<>();
+        List<String> fixingStepNames = fixing.getStepNames();
+        assertThat(fixing.getNbSteps(), is(expectedFixingStepNames.size()));
+        assertThat(fixing.getStepNames().size(), is(expectedFixingStepNames.size()));
+        assertThat(fixing.getStepDurations().size(), is(expectedFixingStepNames.size()));
+        assertThat(fixingStepNames.containsAll(expectedFixingStepNames), is(true));
+    }
+
+    @Test
+    public void testBearsJsonFileWithFailingBuild() throws IOException, ProcessingException {
+        long buggyBuildCandidateId = 208897371; // https://travis-ci.org/surli/failingProject/builds/208897371
+
+        tmpDir = Files.createTempDirectory("test_bears_json_file_failing_build").toFile();
+
+        Build buggyBuildCandidate = this.checkBuildAndReturn(buggyBuildCandidateId, false);
+
+        BuildToBeInspected buildToBeInspected = new BuildToBeInspected(buggyBuildCandidate, null, ScannedBuildStatus.ONLY_FAIL, "test");
+
+        RepairnatorConfig config = RepairnatorConfig.getInstance();
+        config.setLauncherMode(LauncherMode.REPAIR);
+        config.setRepairTools(new HashSet<>(Arrays.asList("Nopol")));
+
+        ProjectInspector inspector = new ProjectInspector(buildToBeInspected, tmpDir.getAbsolutePath(), null, null);
+        inspector.run();
+
+        // check bears.json against schema: this fails since the bears schema was designed for Bears, thus...
```

Comment:
What's the failure?

Reply:
Could you add a FIXME comment: this should led to the creation of a repairnator-schema based on the bears one and then we should be able to check that too.

Reply:
Sure.